src/com/hms_networks/americas/sc/localdatafiles/LocalDataFileManager.java

```diff
@@ -0,0 +1,476 @@
+    localDataFileWriteBuffer.append(getTagTypeAsString(tagType));
+    localDataFileWriteBuffer.append(LocalDataFileConstants.LOCAL_DATA_FILE_DELIMITER);
+
+    // Append tag value and delimiter
+    if (tagType == TagType.STRING) {
+      localDataFileWriteBuffer.append("\"").append(tagValue).append("\"");
+    } else {
+      localDataFileWriteBuffer.append(tagValue);
+    }
+    localDataFileWriteBuffer.append(LocalDataFileConstants.LOCAL_DATA_FILE_DELIMITER);
+
+    // Append data quality and delimiter (not applicable, use filler)
+    localDataFileWriteBuffer.append(LocalDataFileConstants.LOCAL_DATA_FILE_FILLER_VALUE);
+    localDataFileWriteBuffer.append(LocalDataFileConstants.LOCAL_DATA_FILE_DELIMITER);
+
+    // Append alarm type and delimiter
+    localDataFileWriteBuffer.append(alarmType);
+    localDataFileWriteBuffer.append(LocalDataFileConstants.LOCAL_DATA_FILE_DELIMITER);
+
+    // Append alarm status
+    localDataFileWriteBuffer.append(alarmStatus);
+    localDataFileWriteBuffer.append(LocalDataFileConstants.LOCAL_DATA_FILE_LINE_SEPARATOR);
+  }
+
+  /**
+   * Gets a string representation of the specified tag type.
+   *
+   * @param tagType tag type object
+   * @return tag type string
+   */
+  public static String getTagTypeAsString(TagType tagType) {
+    String tagTypeString = LocalDataFileConstants.LOCAL_DATA_FILE_DATA_TYPE_UNKNOWN;
+    if (tagType == TagType.BOOLEAN) {
+      tagTypeString = LocalDataFileConstants.LOCAL_DATA_FILE_DATA_TYPE_BOOLEAN;
+    } else if (tagType == TagType.DWORD) {
+      tagTypeString = LocalDataFileConstants.LOCAL_DATA_FILE_DATA_TYPE_DWORD;
+    } else if (tagType == TagType.FLOAT) {
+      tagTypeString = LocalDataFileConstants.LOCAL_DATA_FILE_DATA_TYPE_FLOAT;
+    } else if (tagType == TagType.INTEGER) {
+      tagTypeString = LocalDataFileConstants.LOCAL_DATA_FILE_DATA_TYPE_INTEGER;
+    } else if (tagType == TagType.STRING) {
+      tagTypeString = LocalDataFileConstants.LOCAL_DATA_FILE_DATA_TYPE_STRING;
+    }
+    return tagTypeString;
+  }
+
+  /**
+   * Gets a string representation of the specified data type.
+   *
+   * @param dataType data type object
+   * @return data type string
+   */
+  public static String getDataTypeAsString(DataType dataType) {
+    String dataTypeString = LocalDataFileConstants.LOCAL_DATA_FILE_DATA_TYPE_UNKNOWN;
+    if (dataType == DataType.BOOLEAN) {
+      dataTypeString = LocalDataFileConstants.LOCAL_DATA_FILE_DATA_TYPE_BOOLEAN;
+    } else if (dataType == DataType.DWORD) {
+      dataTypeString = LocalDataFileConstants.LOCAL_DATA_FILE_DATA_TYPE_DWORD;
+    } else if (dataType == DataType.FLOAT) {
+      dataTypeString = LocalDataFileConstants.LOCAL_DATA_FILE_DATA_TYPE_FLOAT;
+    } else if (dataType == DataType.INTEGER) {
+      dataTypeString = LocalDataFileConstants.LOCAL_DATA_FILE_DATA_TYPE_INTEGER;
+    } else if (dataType == DataType.STRING) {
+      dataTypeString = LocalDataFileConstants.LOCAL_DATA_FILE_DATA_TYPE_STRING;
```

Comment:
switch?

Reply:
Incompatible types. Found: 'com.hms_networks.americas.sc.datapoint.DataType', required: 'byte, char, short or int'

(or Enum in Java 1.5+ as well)